extensions/vim/plugin/bloc-plugin.vim

```diff
@@ -0,0 +1,78 @@
+" bloc generator
+" 1 - Mappings
+nnoremap K :BlocPlugin
+nnoremap K :CubitPlugin
+
+function SearchAndReplace(...)
+        let search = a:000[0]
+        let replace = a:000[1]
+        let type = a:000[2]
+        let path = a:000[3]
+        let fileName = a:000[4]
+
+        let stringCommand = "sed -i'.bak' 's/" . search . "/" . replace . "/gi' " . path . "/" . type . "/" . fileName
+        call system(stringCommand)
+endfunction
+
+function RenameAndDeleteFiles(...)
+        let allFiles =  split(a:000[0])
+        let type = a:000[1]
+        let name = a:000[2]
+        let path = a:000[3]
+        let snakeCaseName = substitute(name, '\u', '_\l&', "g")[1:]
+
+        for i in allFiles
+                call SearchAndReplace("<rename_file>", snakeCaseName, type, path, i)
+                call SearchAndReplace("<rename>", name, type, path, i)
+
+                let destinationName = snakeCaseName . "_" . i
+                let renameSource = "mv " . path . "/" . type . "/" . i . " " . path .  "/" . type . "/" . destinationName
+                call system(renameSource)
+
+                let deleteFile = "rm " . path . "/" . type . "/" . i . ".bak"
+                call system(deleteFile)
+        endfor
+
+        let remoGit = "rm -rf " . path . "/" . type . "/.git"
+        call system(remoGit)
+
+endfunction
+
+function! BlocPlugin(...)
+        let args = split(a:000[0])
+
+        if len(args) < 2
+                echo "You must pass the BlocName and destinationPath like ':Bloc BlocName lib/feature_Name'"
+        else
+                let blocName = args[0]
+                let path = args[1]
+                let command = "git clone https://github.com/eliasreis54/vim_bloc_plugin_source.git " . path . "/bloc/"
```

Comment:
Should this be moved to the bloc mono repo as well?

Reply:
Yes! I think it is the best way to do that.
Can i do it in other pull request?

Reply:
IMO It would probably be worth doing on the same one, because that will require you to change this command as well. But @felangel will know what is best in this case.

Reply:
Maybe adding the section on the site mentioning this plugin should be done on the same PR?

Reply:
I've also been meaning to consolidate the templates used by vscode and intelliJ so that they can share a mason template (brick) rather than having to maintain 3 separate templates but that can be left for a future PR 😄